sdks/python/apache_beam/coders/coder_impl.py

```diff
@@ -161,10 +161,10 @@ def get_estimated_size_and_observables(self, value, nested=False):
       return self.estimate_size(value, nested), []
 
 
-class DeterministicPickleCoderImpl(CoderImpl):
+class DeterministicCoderImpl(CoderImpl):
```

Comment:
I think the name "DeterministicCoderImpl" is misleading as it almost implies that all deterministic CoderImpls should subclass from this.  We can do something like "DeterministicObjectCoderImpl" (not quite satisfied with that particular name...)

+@robertwb for comment